src/core/server/ui_settings/create_or_upgrade_saved_config/create_or_upgrade_saved_config.ts

```diff
@@ -39,10 +39,46 @@ export async function createOrUpgradeSavedConfig(
     version,
   });
 
+  let defaultIndex: string | undefined;
```

Comment:
The implementation looks like it'll do the job (I haven't pulled the code yet).
The following comments are more around trying to be consistent but shouldn't hold this PR from being merged:

The special handling of the `defaultIndex` attribute within the `config` saved object is a transform function, similar to the way other saved objects transform parts of their saved object doc during migrations. 

Even though we can't apply the same principles as for other saved objects, what do you think about extracting the transformation into an attribute-transform function that we apply here?

I'm not too sure where the transform function itself should go within the code folder structure though. We deal with the [whole config object](https://github.com/elastic/kibana/blob/main/src/core/server/ui_settings/create_or_upgrade_saved_config/create_or_upgrade_saved_config.ts) in `create_or_upgrade_saved_config` and specific [config document attributes](https://github.com/elastic/kibana/blob/main/src/core/server/ui_settings/saved_objects/migrations.ts) in `saved_objects` (where the migrations _should_ technically be located). Ideally, the `defaultIndex` transform should belong with the other migrations even if we call it `createOrUpgradeSavedConfig`. WDYT?

Reply:
Excellent idea! I made the changes in 110e043915db7be8c7f008a90dd36763103bd9cc